util/file_reader_writer.cc

```diff
@@ -797,9 +797,12 @@ bool FilePrefetchBuffer::TryReadFromCache(uint64_t offset, size_t n,
     if (readahead_size_ > 0) {
       assert(file_reader_ != nullptr);
       assert(max_readahead_size_ >= readahead_size_);
-
-      Status s =
-          Prefetch(file_reader_, offset, n + readahead_size_, for_compaction);
+      Status s;
+      if (for_compaction) {
+        s = Prefetch(file_reader_, offset, readahead_size_, for_compaction);
```

Comment:
This probably is what caused problem. It should be `std::max(n, readahead_size_)`